c_glib/test/gandiva/test-projector.rb

```diff
@@ -0,0 +1,46 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+class TestProjector < Test::Unit::TestCase
+  include Helper::Buildable
+
+  def setup
+    omit("Gandiva is required") unless defined?(::Gandiva)
+  end
+
+  def test_evaluate
+    field1 = Arrow::Field.new("field1", Arrow::Int32DataType.new)
+    field2 = Arrow::Field.new("field2", Arrow::Int32DataType.new)
+    schema = Arrow::Schema.new([field1, field2])
+    field_sum = Arrow::Field.new("add", Arrow::Int32DataType.new)
+    field_subtract = Arrow::Field.new("subtract", Arrow::Int32DataType.new)
+    sum_expression = Gandiva::Expression.new("add", [field1, field2], field_sum)
+    subtract_expression = Gandiva::Expression.new("subtract", [field1, field2], field_subtract)
+    projector = Gandiva::Projector.new(schema, [sum_expression, subtract_expression])
+    input_columns = [
```

Comment:
`input_arrays` may be better because we use "column" as "field + array".
(It's better that we update `columns` in `garrow_record_batch_new()`.)

Reply:
I've changed it.
I'll send follow-up pull requests to update `columns` in `garrow_record_batch_new()`.